srtcore/api.h

```diff
@@ -239,17 +244,70 @@ friend class CRendezvousQueue;
 
    CUDTException* getError();
 
+   CUDTGroup& addGroup(SRTSOCKET id)
+   {
+       srt::sync::CGuard cg (m_GlobControlLock);
+       // This only ensures that the element exists.
+       // If the element was newly added, it will be NULL.
+       CUDTGroup*& g = m_Groups[id];
+       if (!g)
+       {
+           // This is a reference to the cell, so it will
+           // rewrite it into the map.
+           g = new CUDTGroup;
+       }
+
+       // Now we are sure that g is not NULL,
+       // and persistence of this object is in the map.
+       // The reference to the object can be safely returned here.
+       return *g;
+   }
+
+   void deleteGroup(CUDTGroup* g)
+   {
+       using srt_logging::mglog;
+
+       srt::sync::CGuard cg (m_GlobControlLock);
+
+       CUDTGroup* pg = map_get(m_Groups, g->m_GroupID, NULL);
+       if (pg)
+           delete pg;
```

Comment:
`if (pg)` not needed

Reply:
This condition doesn't check if `pg` is capable of deletion, but whether the group has been found in the container. The group is to be deleted anyway. This check searches back for the groups in the group container by ID, but it should be exactly the found group deleted, which **should** be the same group as the one passed by the pointer, but some IPE might needed to be reported also for that case.